### ISSUE TITLE
No item drops yet?

### ISSUE THREAD

**Issue by nitoygo** [MUnique/OpenMU#118]
I am trying to test out some basic features of this game, and I noticed that monsters don't drop items...
Is this working on your environment?

If this is expected to be working already, could anyone help me how to set it up?

**Comment by sven-n** [comment]
Uhm, I’m not sure but I think the configuration is missing or drop rate is extremely low. The logic should already support item drops (ItemDropGenerator).
I’ll be able to check it in a few days, so please don’t wonder about the slow progress ;)

**Comment by nitoygo** [comment]
> Uhm, I’m not sure but I think the configuration is missing or drop rate is extremely low. The logic should already support item drops (ItemDropGenerator).
I am checking the tables in openmu database, I can't seem to find such config. (or maybe I just missed it)
> I’ll be able to check it in a few days, so please don’t wonder about the slow progress ;)
Thanks! No worries 👍

**Comment by nitoygo** [comment]
Btw I tried hard coding the DefaultDropGenerator to always drop a specific type (e.g. Excellent), it worked, but, I can't pick it up. 

I tried dropping some items from my inventory and I can't pick them up too.

I will try to investigate this tomorrow. If you can guide me where to look, or you already have an idea, I'd appreciate it.

**Comment by sven-n** [comment]
Item drops are configured in “DropItemGroups”. I think I didn’t add them yet to the monsters in the Initialization project.

**Comment by sven-n** [comment]
I fixed the data initialization, it should drop items now, too.
Reopened, because there might be a pickup issue left. A starting point for debugging is here the PickupItemAction.

**Comment by nitoygo** [comment]
"DropItemGroups" is now updated after the last update. But monsters still does not drop items.

> Reopened, because there might be a pickup issue left. A starting point for debugging is here the PickupItemAction.

Is this about [ths comment](https://github.com/MUnique/OpenMU/issues/120#issuecomment-496242997) ?

**Comment by sven-n** [comment]
I found two issues:
1. ItemDropGroups are not loaded correctly when the server is started. It's a bit complicated to fix and I will need a bit more time ;)
2. The DropGenerator creates temporary item objects. When added to the inventory they get converted. Unfortunately, the code did nothing with the converted item. I fixed that.

**Comment by nitoygo** [comment]
> 2. The DropGenerator creates temporary item objects. When added to the inventory they get converted. Unfortunately, the code did nothing with the converted item. I fixed that.

Hi! I was able to test this but I encountered an exception here (in TryPickUp() method)
https://github.com/MUnique/OpenMU/blob/master/src/GameLogic/DroppedItem.cs#L175
```C#
    player.PersistenceContext.Attach(this.Item);
```

(Something about entity framework exception...)

Although the InventoryStorage.AddItem() was already modified to make the temporary item persistent, the item instance in DroppedItem class is still not.

Sorry I'm not familiar yet with entity framework but, is the above line still needed? 
Or simply calling ```player.Inventory.AddItem(...this.Item)``` is enough?
Or ```player.PersistenceContext.Attach(this.Item);``` should be called inside ```InventoryStorage.AddItem()``` (where the converted item instance is known) instead?

Thanks!
